tests/tests_default.yml

```diff
@@ -4,10 +4,32 @@
   hosts: all
   gather_facts: false  # test that role works in this case
   roles:
-    - linux-system-roles.template
+    - linux-system-roles.keylime_server
   tasks:
+    - name: Check header for ansible_managed, fingerprint - legacy
+      include_tasks: tasks/check_header.yml
+      vars:
+        __file: /etc/keylime.conf
+        __fingerprint: system_role:keylime_server
+      when: __keylime_server_legacy_config
```

Comment:
Are you trying to use the value of `__keylime_server_legacy_config` defined in `vars/`?  If so, it will be undefined - vars defined in `vars/` are not exported.

Instead, you have to use `include_role` with the `public: true` modifier to make all of the role private variables exposed to the playbook:

```yaml
  tasks:
    - name: Run the role
      include_role:
        name: linux-system-roles.keylime_server
      public: true
```

Then you can use all of the role private variables such as `__keylime_server_legacy_config`, etc. in the test playbook.

Reply:
Alright, let me try this.